crates/admin/protobuf/cluster_ctrl_svc.proto

```diff
@@ -85,11 +85,24 @@ message TrimLogRequest {
   uint64 trim_point = 2;
 }
 
-message CreatePartitionSnapshotRequest { uint32 partition_id = 1; }
+// Requests a partition snapshot to be archived. Optionally, a minimum
+// target LSN for the snapshot may be set and the log can be trimmed to the
+// archived LSN at the same time. It is safe to retry this operation on
+// error with the same inputs.
+message CreatePartitionSnapshotRequest {
+  uint32 partition_id = 1;
+  // Minimum LSN (inclusive) which must be covered by the snapshot; if set it is treated as a hard
+  // requirement and the operation will return an error if it cannot be met
+  optional uint64 min_target_lsn = 2;
+  // Optionally, also trim the log to the archived LSN; trim only occurs after a successful snapshot
+  // and so will honor the min_target_lsn requirement, if set
+  optional bool trim_log = 3;
```

Comment:
Why using `optional` with a bool primitive type?

Reply:
I thought it's more self-documenting this way, but reasonable suggestion not to!

Reply:
updated :-)

Reply:
This is not what I mean. I mean why using the keyword `optional` when a bool can only be true or false. In other words, what is the difference between `none` and `false` ?